crates/asm/worker/src/handle.rs

```diff
@@ -1,24 +1,39 @@
 use async_trait::async_trait;
 use strata_primitives::prelude::*;
-use strata_service::CommandHandle;
+use strata_service::{CommandHandle, ServiceMonitor};
 use strata_state::BlockSubmitter;
 use tracing::warn;
 
+use crate::{AsmWorkerService, WorkerContext};
+
+type Monitor<W> = ServiceMonitor<AsmWorkerService<W>>;
+
 /// Handle for interacting with the ASM worker service.
 #[derive(Debug)]
-pub struct AsmWorkerHandle {
+pub struct AsmWorkerHandle<W: WorkerContext + Send + Sync + 'static> {
     command_handle: CommandHandle<L1BlockCommitment>,
+    service_monitor: Monitor<W>,
```

Comment:
This is super ugly, I should have converted this like I did the command handle type to just use the status type, so that the monitor type doesn't have to itself be generic over the context (and make all the consumers be aware of the concrete context type).  It's fun and quirky to have it be generic over the service but it's actually cumbersome when the service itself is generic.  Maybe I should have reworked how we'd use a context trait instead, but that sounds like it's getting close to needing HKTs.

Reply:
I vaguely plan on working around "integrating CSM with ASM logs" later, so I'll do how it's done in CommandHandle, this will improve the ergonomics a bit.